src/lumigo_tracer/utils.py

```diff
@@ -161,3 +164,71 @@ def is_aws_environment():
     :return: heuristically determine rather we're running on an aws environment.
     """
     return bool(os.environ.get("LAMBDA_RUNTIME_DIR"))
+
+
+def extract_frames_from_exception() -> List[dict]:
+    """
+    Should be called only in `except` scope.
+    """
+    free_space = MAX_VARS_SIZE
+    frames: List[dict] = []
+    frames_infos = inspect.trace()
+    for frame_info in reversed(frames_infos):
+        if free_space <= 0 or "lumigo_tracer" in frame_info.filename:
+            return frames
+        frames.append(convert_frame(frame_info, free_space))
+        free_space -= len(json.dumps(frames[-1]))
+    return frames
+
+
+def convert_frame(frame_info: inspect.FrameInfo, free_space: int) -> dict:
+    return {
+        "lineno": frame_info.lineno,
+        "fileName": frame_info.filename,
+        "function": frame_info.function,
+        "variables": _truncate_locals(frame_info.frame.f_locals, free_space),
+    }
+
+
+def _truncate_locals(f_locals: Dict[str, Any], free_space: int) -> FrameVariables:
```

Comment:
you have the same logic at `_create_request_body` 
create a testable generic function that gets a list and size and returns pruned list